app/packages/playback/src/lib/state.ts

```diff
@@ -0,0 +1,384 @@
+import { atom } from "jotai";
+import { atomFamily } from "jotai/utils";
+import { LRUCache } from "lru-cache";
+import { BufferManager, BufferRange } from "../../../utilities/src";
+import {
+  ATOM_FAMILY_CONFIGS_LRU_CACHE_SIZE,
+  DEFAULT_FRAME_NUMBER,
+  DEFAULT_LOOP,
+  DEFAULT_SPEED,
+  DEFAULT_TARGET_FRAME_RATE,
+  DEFAULT_USE_TIME_INDICATOR,
+  LOAD_RANGE_SIZE,
+} from "./constants";
+
+export type PlayheadState =
+  | "playing"
+  | "paused"
+  | "waitingToPlay"
+  | "waitingToPause";
+
+export type TimelineName = string;
+export type FrameNumber = number;
+export type TargetFrameRate = number;
+export type Speed = number;
+export type TotalFrames = number;
+export type TimelineSubscribersMap = Map<
+  SubscriptionId,
+  SequenceTimelineSubscription
+>;
+
+// tood: think about making it a symbol and subscribers a WeakMap
+export type SubscriptionId = string;
```

Comment:
Firefox does not yet support symbols as keys in WeakMaps 😞 
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap#browser_compatibility

Spotlight uses this `ID` type as an alternative
* https://github.com/voxel51/fiftyone/blob/6e26d52b16671846a6a9f986f1ad3e3357384360/app/packages/spotlight/src/types.ts#L29